app/lib/features/space/sheets/link_room_sheet.dart

```diff
@@ -0,0 +1,422 @@
+import 'package:acter/common/providers/chat_providers.dart';
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/widgets/default_button.dart';
+import 'package:acter/common/widgets/search.dart';
+import 'package:acter/common/widgets/side_sheet.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter/features/home/widgets/space_chip.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+// ChildRoomType configures the sub child type of the `Spaces`
+enum ChildRoomType {
+  chat,
+  space,
+  recommendedSpace,
+}
+
+class LinkRoomPage extends ConsumerStatefulWidget {
+  final String? parentSpaceId;
+  final String? childRoomType;
+
+  const LinkRoomPage({
+    super.key,
+    required this.parentSpaceId,
+    required this.childRoomType,
+  });
+
+  @override
+  ConsumerState<LinkRoomPage> createState() => _LinkRoomPageConsumerState();
+}
+
+class _LinkRoomPageConsumerState extends ConsumerState<LinkRoomPage> {
+  ChildRoomType childRoomType = ChildRoomType.space;
+  String sheetHeader = '';
+  final TextEditingController searchTextEditingController =
+      TextEditingController();
+  final List<String> childRoomsIds = [];
+  final List<String> recommendedChildSpaceIds = [];
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      initData();
+      selectParentSpaceData();
+      fetchKnownSubChatsData();
+    });
+  }
+
+  //Select parent space data
+  void initData() {
+    //Set data based on the childRoomType
+    if (widget.childRoomType == ChildRoomType.chat.name) {
+      childRoomType = ChildRoomType.chat;
+      sheetHeader = 'Link as Space-chat';
+    } else if (widget.childRoomType == ChildRoomType.space.name) {
+      childRoomType = ChildRoomType.space;
+      sheetHeader = 'Link Sub-Space';
+    } else if (widget.childRoomType == ChildRoomType.recommendedSpace.name) {
+      childRoomType = ChildRoomType.recommendedSpace;
+      sheetHeader = 'Link Recommended-Space';
+    } else {
+      childRoomType = ChildRoomType.space;
+      sheetHeader = 'Link Sub-Space';
+    }
+  }
+
+  //Select parent space data
+  void selectParentSpaceData() {
+    if (widget.parentSpaceId != null) {
+      final notifier = ref.read(selectedSpaceIdProvider.notifier);
+      notifier.state = widget.parentSpaceId;
+    }
```

Comment:
It's never optional.

```suggestion
    final notifier = ref.read(selectedSpaceIdProvider.notifier);
    notifier.state = widget.parentSpaceId;
```